### PR TITLE
[FEATURE] more CI platforms and fix of current bugs

### PR DESCRIPTION
**Pull request by h-2**
there are currently issues in travis. Moving to the trusty images hopefully will fix it. Also I have added more compilers and osx.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,28 +1,55 @@
...
+    - os: linux
+      compiler: gcc-4.9
+      addons:
+        apt:
+          sources: ['ubuntu-toolchain-r-test']
+          packages: ['g++-4.9', 'cmake', 'zlib1g-dev', 'libbz2-dev', 'libboost-dev', 'python', 'python-nose', 'python-jinja2', 'python-pip']
+      install: ['export CXX="g++-4.9" CC="gcc-4.9"' ]
+
+    - os: linux
+      compiler: gcc-4.9
```
Comment (rrahn) [inline .travis.yml]: schould be 4.8?

**Comment by rrahn** [comment]
Is it possible to run the builds in parallel? Right now it seems they are in a queue waiting until the previous job is done. That would not be a desired behaviour.

**Comment by h-2** [comment]
> Right now it seems they are in a queue waiting until the previous 

Travis is just throttling our access to VMs because we are requesting so many builds at the moment. Not related to the changes. However the new VMs seem to be limited to 50min which is a problem, since our builds take 55min :unamused: 
I have hopes for the ccache to speed it up, but I am not yet sure it works on travis, so hold this please.

**Comment by h-2** [comment]
superseded by #1346

**Pull request closed without merging**